src/UndockedRegFreeWinRT/UndockedRegFreeWinRT/catalog.cpp

```diff
@@ -81,72 +82,373 @@ struct component
+    {
+        this_component->threading_model = ABI::Windows::Foundation::ThreadingType::ThreadingType_BOTH;
+    }
+    else
+    {
+        return HRESULT_FROM_WIN32(ERROR_SXS_MANIFEST_PARSE_ERROR);
+    }
+
+    if (activatableClass == nullptr || !activatableClass[0])
+    {
+        return HRESULT_FROM_WIN32(ERROR_SXS_MANIFEST_PARSE_ERROR);
+    }
+    this_component->xmlns = xmlns; // Should we care if this value is blank or missing?
+    // Check for duplicate activatable classes
+    auto component_iter = g_types.find(activatableClass);
+    if (component_iter != g_types.end()) 
+    {
+        return HRESULT_FROM_WIN32(ERROR_SXS_DUPLICATE_ACTIVATABLE_CLASS);
+    }
+    g_types[activatableClass] = this_component;
+    return S_OK;
+}
 
+HRESULT ParseDependentAssemblyTag(ComPtr<IXmlReader> xmlReader)
+{
+    XmlNodeType nodeType;
+    LPCWSTR localName = nullptr;
+    auto locale = _create_locale(LC_ALL, "C");
+    while (S_OK == xmlReader->Read(&nodeType))
+    {
+        if (nodeType == XmlNodeType_Element)
+        {
+            RETURN_IF_FAILED((xmlReader->GetLocalName(&localName, nullptr)));
+            if (_wcsicmp_l(localName, L"assemblyIdentity", locale) == 0)
+            {
+                RETURN_IF_FAILED(ParseAssemblyIdentityTag(xmlReader, true));
+            }
+        }
+        else if (nodeType == XmlNodeType_EndElement)
+        {
+            RETURN_IF_FAILED((xmlReader->GetLocalName(&localName, nullptr)));
+            if (_wcsicmp_l(localName, L"dependentAssembly", locale) == 0)
+            {
+                return S_OK;
+            }
+        }
+    }
     return S_OK;
 }
 
+HRESULT ParseAssemblyIdentityTag(ComPtr<IXmlReader> xmlReader, bool searchDependentAssembly)
+{
+    HRESULT hr = S_OK;
+    LPCWSTR dependentAssemblyFileName = nullptr;
+    hr = xmlReader->MoveToAttributeByName(L"name", nullptr);
+    if (hr != S_OK)
+    {
+        return HRESULT_FROM_WIN32(ERROR_SXS_MANIFEST_PARSE_ERROR);
+    }
+    RETURN_IF_FAILED(xmlReader->GetValue(&dependentAssemblyFileName, nullptr));
+    if (dependentAssemblyFileName == nullptr || !dependentAssemblyFileName[0])
+    {
+        return HRESULT_FROM_WIN32(ERROR_SXS_MANIFEST_PARSE_ERROR);
+    }
+    // Look out for assemblies that have already been parsed so we don't have circular dependencies
+    auto itr = g_assemblies.find(dependentAssemblyFileName);
+    if (itr != g_assemblies.end())
+    {
+        return S_OK; // We've already look through this assembly, no need to look insert or search for it again. 
+    }
+    g_assemblies.insert(dependentAssemblyFileName);;
+    if (searchDependentAssembly)
+    {
+        // Don't fail if we don't find the assembly, in the realworld, manifests may have 
+        // dependentAssembly ref to WinSxS assemblies which this code won't find.
+        SearchAssembly(dependentAssemblyFileName);
+    }
+    return hr;
+}
+
+
+HRESULT SearchAssembly(std::wstring dependentAssemblyFileName)
```

Comment:
wstring_view ? Or `PCWSTR` since that's mostly what you're calling it woth?

Reply:
Deleted this method